src/Permissions/PermissionSetter.php

```diff
@@ -116,8 +117,9 @@ public function setPermissions($baseDir, $targetDir)
     private function resolvePattern($baseDir, $targetDir, $pattern)
     {
         if (strpos($pattern, '*') === false) {
+            $basePath = $baseDir . '/' . $pattern;
             $targetPath = $targetDir . '/' . $pattern;
-            if (!file_exists($targetPath)) {
+            if (!file_exists($basePath) || !file_exists($targetPath)) {
```

Comment:
This was the problem area. It was always returning the path to be set even if it didn't exist in the base path (package). Tests added for this scenario.